### PR TITLE
Fix Stack Exchange parsing

### PR DESCRIPTION
**Pull request by pcallier**
Generate "clusters" of size 2 from Stack Exchange data, labeling "related" posts as {novel, novel} and "duplicate" posts as {novel, redundant}. Adds some logic to allow for picking up parsing jobs in midstream if they're cut short or their intermediate products (unzipped Stack Exchange dumps, etc) are already available.

### CODE REVIEW

File: src/data/stackexchange/sitelist.txt [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+vi
+webapps
+webmasters
+windowsphone
+woodworking
+wordpress
+workplace
+worldbuilding
+writers
+askubuntu.com
```
Comment (dave-lab41) [inline src/data/stackexchange/sitelist.txt]: I believe that there are a few more .com entries, including superuser.com

File: src/data/stackexchange/stack_exchange_parse.py [hunk 2]
```diff
@@ -278,30 +224,36 @@ def main(args):
 
     if args.skipparse == False:
```
Comment (dave-lab41) [inline src/data/stackexchange/stack_exchange_parse.py]: I believe your new logic of verifying completed parsing components removes the need to have a skipparse command line argument

File: src/data/stackexchange/stack_exchange_parse.py [hunk 3]
```diff
@@ -278,30 +224,36 @@ def main(args):
...
 
             #writes cluster information to json files
-            write_json_files(clusters, related, duplicates, corpus, corpus_section_directory)
+            write_json_files(clusters, corpus_section_directory)
+            
+            # put completion marker in folder so we can skip it next time
+            with open(os.path.join(corpus_section_directory, ".done"), "w") as f:
+                print("", file=f)
 
             print("Completed " + section)
```
Comment (dave-lab41) [inline src/data/stackexchange/stack_exchange_parse.py]: The next step was built to filter the json files to only retain clusters that had between 3 and 10 posts. Given the new paired posting logic, I believe that the filter command line argument and the filter_json_files function are no longer needed.

**Comment by pcallier** [comment]
All right, I think that's better. I used Kim's list of sites, which was longer and had superuser and some others.

**Pull request merged**